core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1035,6 +1035,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     postEnvironmentUpdate()
   }
 
+  private[spark] def supportKillExecutor = master.contains("yarn") || dynamicAllocationTesting
```

Comment:
this is also used in `requestExecutors` so we need to generalize the name. Also, please add a java doc
```
/**
 * Return whether dynamically adjusting the amount of resources allocated to
 * this application is supported. This is currently only available for YARN.
 */
private[spark] def supportDynamicAllocation = master.contains("yarn") || dynamicAllocationTesting
```